conference/accounts.py

```diff
@@ -51,12 +68,253 @@ def clean(self):
+            raise forms.ValidationError("Sorry, that's a wrong answer")
+        return self.cleaned_data['captcha_question']
+
+    def clean_email(self):
+        email = self.cleaned_data['email']
+        if User.objects.filter(email__iexact=email).count() > 0:
+            raise forms.ValidationError('Email already in use')
+
+        return email.lower()
+
+    def clean(self):
+        if not self.cleaned_data.get('i_accept_privacy_policy'):
+            raise forms.ValidationError(PRIVACY_POLICY_ERROR)
+
+        if not self.is_valid():
+            return super().clean()
+
+        data = self.cleaned_data
+        if data['password1'] != data['password2']:
+            raise forms.ValidationError('password mismatch')
+        return data
+
+
 urlpatterns = [
     url(
         r"^login/$",
         auth_views.LoginView.as_view(
             authentication_form=LoginForm, template_name=LOGIN_TEMPLATE
         ),
         name="login",
-    )
+    ),
+    url(r"^logout/$", auth_views.LogoutView.as_view(), name="logout"),
+    url(
+        r"^signup/$",
+        signup_step_1_create_account,
+        name="signup_step_1_create_account",
+    ),
+    url(
+        r"^signup/thanks/$",
+        signup_step_2_please_verify_email,
+        name="signup_step_2_please_verify_email",
+    ),
+    url(
+        # 22 not 36 because we use short uuid
+        r"^signup/verify-email/(?P<token>\w{22})/$",
+        handle_verification_token,
+        name="handle_verification_token",
+    ),
+    # Password reset, using default django views.
+    url(
+        r"^password-reset/$",
+        auth_views.password_reset,
+        kwargs={
+            "template_name": "ep19/bs/accounts/password_reset.html",
+            "post_reset_redirect": reverse_lazy(
+                "accounts:password_reset_done"
+            ),
+            "email_template_name": "ep19/emails/password_reset_email.txt",
+            "subject_template_name": "ep19/emails/password_reset_subject.txt",
+        },
+        name="password_reset",
+    ),
+    url(
+        r"^password-reset/done/$",
+        auth_views.password_reset_done,
+        kwargs={"template_name": "ep19/bs/accounts/password_reset_done.html"},
+        name="password_reset_done",
+    ),
+    url(
+        r"^reset/(?P<uidb64>[\w-]+)/(?P<token>[\w]{1,13}-[\w]{1,20})/$",
+        auth_views.password_reset_confirm,
+        kwargs={
+            "set_password_form": auth_forms.SetPasswordForm,
+            "template_name": "ep19/bs/accounts/password_reset_confirm.html",
+            "post_reset_redirect": reverse_lazy(
+                "accounts:password_reset_complete"
+            ),
+        },
+        name="password_reset_confirm",
+    ),
+    url(
+        r"^reset/done/$",
+        auth_views.password_reset_complete,
+        kwargs={
+            "template_name": "ep19/bs/accounts/password_reset_complete.html"
+        },
+        name="password_reset_complete",
+    ),
 ]
```

Comment:
organising by use case vs organising by type of code that lives in a file.

In general I'm a fan of how django sets things up, but in this particular case we have a lot of weird organisation and dependencies in the "old" code, so I opted for a module-per-use-case based approach, to make things easier to find later on.

It should be a temporary situation, because ideally all those modules (like `conference/invoicing.py, conference/cfp.py conference/cart.py, etc) – should  probably be separate django apps, however django apps should also contain their own models, and I didn't want to move models too much for this iteration of updates because that would create too much noise.